test/distributed/org/apache/cassandra/distributed/test/ShortReadProtectionTest.java

```diff
@@ -66,51 +329,192 @@ public void testGroupBySRPRegularRow() throws Throwable
+                         "INSERT INTO %s (pk, ck) VALUES (2, 0) USING TIMESTAMP 42",
+                         "INSERT INTO %s (pk, ck) VALUES (2, 1) USING TIMESTAMP 42",
+                         "INSERT INTO %s (pk, ck) VALUES (4, 0) USING TIMESTAMP 42",
+                         "INSERT INTO %s (pk, ck) VALUES (4, 1) USING TIMESTAMP 42")
+              .assertRows("SELECT pk, ck FROM %s PER PARTITION LIMIT 2 LIMIT 4",
+                          row(0, 0), row(0, 1), row(4, 0), row(4, 1));
+    }
+
+    private static long token(int key)
+    {
+        return (long) Murmur3Partitioner.instance.getToken(ByteBufferUtil.bytes(key)).getTokenValue();
+    }
+
+    private static class Tester
+    {
+        private static final AtomicInteger seqNumber = new AtomicInteger();
+
+        private final ICoordinator coordinator;
+        private final boolean quorum, flush, paging;
+        private final String qualifiedTableName;
+        private final ConsistencyLevel consistencyLevel;
+
+        private boolean flushed = false;
+
+        private Tester(boolean quorum, boolean flush, boolean paging, int coordinator)
+        {
+            this.coordinator = cluster.coordinator(coordinator);
+            this.quorum = quorum;
+            this.flush = flush;
+            this.paging = paging;
+            qualifiedTableName = KEYSPACE + ".t_" + seqNumber.getAndIncrement();
+            consistencyLevel = quorum ? QUORUM : ALL;
+        }
+
+        private Tester createTable(String query)
+        {
+            cluster.schemaChange(format(query) + " WITH read_repair='NONE'");
+            return this;
+        }
+
+        private Tester allNodes(int startInclusive, int endExclusive, Function<Integer, String> querySupplier)
+        {
+            IntStream.range(startInclusive, endExclusive).mapToObj(querySupplier::apply).forEach(this::allNodes);
+            return this;
+        }
+
+        private Tester allNodes(String... queries)
+        {
+            for (String query : queries)
+                allNodes(query);
+            return this;
+        }
+
+        private Tester allNodes(String query)
+        {
+            coordinator.execute(format(query), ALL);
+            return this;
+        }
+
+        private Tester onlyNode1(String... queries)
+        {
+            onlyNodeN(1, queries);
+            return this;
+        }
+
+        private Tester onlyNode2(String... queries)
+        {
+            onlyNodeN(2, queries);
+            return this;
+        }
+
+        private Tester onlyNode3(String... queries)
+        {
+            onlyNodeN(3, queries);
+            return this;
+        }
+
+        private void onlyNodeN(int node, String... queries)
+        {
+            for (String query : queries)
+            {
+                String formattedQuery = format(query);
+                cluster.get(node).executeInternal(formattedQuery);
+                if (quorum)
```

Comment:
What if we just kept the `consistencyLevel`, called it `readConsistency` (or similar), and got rid of the `quorum` field? (We could just check read CL here instead of needing `quorum`.)

Reply:
Done. I was using a boolean trying to express that we only use between two read-write consistency models (ONE-ALL and QUORUM-QUORUM), but that way is probably clearer.